src/server/admin-controller.js

```diff
@@ -1,41 +1,89 @@
 import PageRenderer from './page-renderer.js';
 import permissions from '../domain/permissions.js';
 import { PAGE_NAME } from '../domain/page-name.js';
+import { BATCH_TYPE } from '../domain/constants.js';
+import BatchService from '../services/batch-service.js';
 import SettingService from '../services/setting-service.js';
 import StorageService from '../services/storage-service.js';
 import DocumentService from '../services/document-service.js';
 import needsPermission from '../domain/needs-permission-middleware.js';
 import ClientDataMappingService from '../services/client-data-mapping-service.js';
 
 class AdminController {
-  static get inject() { return [SettingService, DocumentService, StorageService, ClientDataMappingService, PageRenderer]; }
+  static get inject() { return [SettingService, DocumentService, StorageService, BatchService, ClientDataMappingService, PageRenderer]; }
 
-  constructor(settingService, documentService, storageService, clientDataMappingService, pageRenderer) {
+  constructor(settingService, documentService, storageService, batchService, clientDataMappingService, pageRenderer) {
     this.settingService = settingService;
     this.documentService = documentService;
     this.storageService = storageService;
+    this.batchService = batchService;
     this.clientDataMappingService = clientDataMappingService;
     this.pageRenderer = pageRenderer;
   }
 
   async handleGetAdminPage(req, res) {
-    const [settings, docs, storagePlans] = await Promise.all([
+    const { user } = req;
+
+    const [
+      settings,
+      documents,
+      storagePlans,
+      lastDocumentRegenerationBatch,
+      lastCdnResourcesConsolidationBatch
+    ] = await Promise.all([
       this.settingService.getAllSettings(),
       this.documentService.getAllDocumentsMetadata(),
-      this.storageService.getAllStoragePlansWithAssignedUserCount()
+      this.storageService.getAllStoragePlansWithAssignedUserCount(),
+      this.batchService.getLastBatch(BATCH_TYPE.documentRegeneration),
+      this.batchService.getLastBatch(BATCH_TYPE.cdnResourcesConsolidation)
     ]);
-    const documents = await this.clientDataMappingService.mapDocsOrRevisions(docs, req.user);
-    const initialState = { settings, documents, storagePlans };
+
+    const initialState = {
+      settings,
+      documents: await this.clientDataMappingService.mapDocsOrRevisions(documents, user),
+      storagePlans,
+      lastDocumentRegenerationBatch: await this.clientDataMappingService.mapBatch(lastDocumentRegenerationBatch, user),
+      lastCdnResourcesConsolidationBatch: await this.clientDataMappingService.mapBatch(lastCdnResourcesConsolidationBatch, user)
```

Comment:
From within the mapper we get an exception when either of `lastDocumentRegenerationBatch` and `lastCdnResourcesConsolidationBatch` is `null`.

Reply:
Oh, I haven't tested that, thanks for spotting it :+1: